packages/blocks/src/api/serializer.js

```diff
@@ -50,6 +51,26 @@ export function getBlockDefaultClassName( blockName ) {
 	);
 }
 
+/**
+ * Returns a block variation specific classname.
+ *
+ * If the given block matches a variation, the classname will be the block's default classname
+ * with the variation name appended (separated by a hyphen).
+ *
+ * @param {string} blockName  The block name.
+ * @param {Object} attributes Block attributes.
+ *
+ * @return {string|null} The block variation classname, or null if the block doesn't match any variation.
+ */
+export function getBlockVariationClassName( blockName, attributes ) {
+	const activeVariation = getActiveBlockVariation( blockName, attributes );
+	if ( ! activeVariation ) {
+		return null;
+	}
+
+	return getBlockDefaultClassName( blockName ) + '-' + activeVariation.name;
```

Comment:
Maybe worth mentioning at the docs that the prefix is from `getBlockDefaultClassName`, which means its value is also affected by the filter? 🤔

Reply:
Yeah, good point!

Reply:
455a7f4f35a

Reply:
And two more notes (right in the JSDoc) here: 19f2e00fdfb

Reply:
I saw a comment from @peterwilsoncc suggesting a different char than `-` as a delimiter in https://github.com/WordPress/gutenberg/pull/61864#pullrequestreview-2152376947. I like that idea as it removes the probability of having the same class for two different blocks and their variations:

`plugin/foo` + `boo` -> `wp-block-plugin-foo-boo`
`plugin/foo-boo` -> `wp-block-plugin-foo-boo`

Reply:
One more thing worth noting here. `getBlockDefaultClassName( blockName )` after filters get applied might change to anything, so it could also be multiple class names: `foo boo bar`. In that case, the last one will get changed extended, and the reset repeated. Should it pick only the first class name returned so that doesn't end up looking as follows:
`foo boo bar foo boo bar-variaiton`

Reply:
> One more thing worth noting here. `getBlockDefaultClassName( blockName )` after filters get applied might change to anything, so it could also be multiple class names: `foo boo bar`. In that case, the last one will get changed extended, and the reset repeated. Should it pick only the first class name returned so that doesn't end up looking as follows: `foo boo bar foo boo bar-variaiton`

Yeah, that's a tricky one. TBH I'm not sure what the answer should be here; multiple options come to mind:
1. Appending the variation to the first class name, as you suggested, i.e. `foo-variation boo bar`.
2. Appending the variation to the last class name only: `foo boo bar-variation`
3. Appending it to the string, splitting it at whitespaces, deduplicating, and rejoining:
   1. Attach: `foo boo bar foo boo bar-variation`
   2. Split: `[ 'foo', 'boo', 'bar', 'foo', 'boo', 'bar-variation' ]`
   3. Deduplicate: `[ 'foo', 'boo', 'bar', 'bar-variation' ]`
   4. Rejoin: `foo boo bar bar-variation`
 This is equivalent to keeping a copy of the last class name, and appending another copy with the variation name attached.

TBH none of these stands out to me as the "right" answer.

My original thinking was to treat the "multiple classnames" case as unsupported (as both the function name `getBlockDefaultClassName` (singular!) and the examples in the docs indicate that it should be _one_ class name only); i.e. people would indeed end up with `foo boo bar foo boo bar-variation`. I'm not sure how much of a problem that would be in practice.

Reply:
Finally, we could consider bypassing the `getBlockDefaultClassName()` function altogether when creating the variation class name, and instead introduce a dedicated one (plus a filter) for the variation.

However, using that function for both the block class name and the variation class name has one big advantage. Consider the following example:

```js
// Our filter function.
function setBlockCustomClassName( className, blockName ) {
	return blockName === 'core/code' ? 'my-plugin-code' : className;
}

// Adding the filter.
wp.hooks.addFilter(
	'blocks.getBlockDefaultClassName',
	'my-plugin/set-block-custom-class-name',
	setBlockCustomClassName
);
```

In this case, if we use `getBlockDefaultClassName` for both the block and variation class name, we get `my-plugin-code my-plugin-code-myvariation` as result.

If we don't use `getBlockDefaultClassName` to generate the variation, we'd get `my-plugin-code wp-code-myvariation`, i.e. we'd lose consistency between the block and variation class names.

Reply:
> I saw a comment from @peterwilsoncc suggesting a different char than `-` as a delimiter in [#61864 (review)](https://github.com/WordPress/gutenberg/pull/61864#pullrequestreview-2152376947). I like that idea as it removes the probability of having the same class for two different blocks and their variations:
> 
> `plugin/foo` + `boo` -> `wp-block-plugin-foo-boo` `plugin/foo-boo` -> `wp-block-plugin-foo-boo`
> 
> Maybe we could follow [BEM rules](https://github.com/WordPress/gutenberg/blob/trunk/docs/contributors/code/coding-guidelines.md#css) and use `__`?

Trying this out in d369a978111.